.github/workflows/flowzone.yml

```diff
@@ -84,6 +84,11 @@ on:
         type: string
         required: false
         default: ""
+      python_versions: 
```

Comment:
Is there a way to specify supported python versions native to python itself, maybe in `pyproject.toml` somehow? If so we could fetch the versions to run against from that file rather than having a separate CI specifier

Reply:
This was my initial attempt but there were two issues which I could not solve. The first one is that I am not sure how to convert multiple inequalities, i.e. `>=3.7,<3.10,!=3.8` into a list for the matrix `[3.7, 3.9]` as poetry often requires the Python versions to be specified with inequalities (packages can become unhappy if a caret is used for the python version in the toml file, here is an [example](https://github.com/python-poetry/poetry/blob/master/docs/faq.md#why-is-poetry-telling-me-that-the-current-projects-python-requirement-is-not-compatible-with-one-or-more-packages-python-requirements)). Just a note that the Go version of `yq` that comes preinstalled on the runner does not support `.toml` files and something like `dasel` needs to be downloaded if we want to parse the .toml properly.

The second one is that poetry does not have something nice like npm check-engine to return a status.

Now when I think about it again the closest thing is probably `poetry install –dry-run` which will return
```
The currently activated Python version 3.8.13 is not supported by the project (~3.9).
Trying to find and use a compatible version.

  NoCompatiblePythonVersionFound
```
I guess checking for `NoCompatiblePythonVersionFound` can be used. I will try with this one to repeat the Node version selection steps.